include/named_threads.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <string>
+#include <thread>
+
+namespace celerity::detail {
+
+void set_thread_name(std::thread& thread, const std::string& name);
```

Comment:
This API could be made more concise with
```c++
std::thread::native_handle_type get_current_thread();
void set_thread_name(std::thread::native_handle_type, const std::string&);
std::string get_thread_name(std::thread::native_handle_type);
```

Then you also wouldn't need the separate `static set/get_thread_name_unix/win` functions in the implementation.

Reply:
Yes, that is a very good idea, thank you.